### PR TITLE
Validate properties values according to database type

### LINKED ISSUE #17683: Incorrect exception message for geoip fields options.
**Elasticsearch version**: master/5.0

**JVM version**: ALL

**OS version**: ALL

**Description of the problem including expected versus actual behavior**:

The ingest-geoip plugin does not distinguish supported `fields` values between city and country 
database types. The error message should only provide suggested valid fields that are compatible with 
the selected database type. Currently, all fields are treated as City-related fields.

The expected behavior should properly distinguish between which fields belong to which database type and provide the appropriate valid fields for the selected database.

Also, `LATITUDE` and `LONGITUDE` are no longer valid fields, these should be removed from the Field enum.
**Steps to reproduce**:
 1. PUT a pipeline definition that includes the following geoip processor definition (`GeoLite2-City.mmdb`):
```
{
  "geoip": {
    "database_file": "GeoLite2-City.mmdb",
    "fields": [ "foo" ]
  }
}
```
you will see the following error message:
```
[fields] illegal field option [test]. valid values are [[IP, COUNTRY_ISO_CODE, COUNTRY_NAME, CONTINENT_NAME, REGION_NAME, CITY_NAME, TIMEZONE, LATITUDE, LONGITUDE, LOCATION]]
```
 2. PUT a pipeline definition that include the following geoip processor definition (`GeoLite2-Country.mmdb`):
```
{
  "geoip": {
    "database_file": "GeoLite2-Country.mmdb",
    "fields": [ "foo" ]
  }
}
```
you will see the following error message:
```
[fields] illegal field option [foo]. valid values are [[IP, COUNTRY_ISO_CODE, COUNTRY_NAME, CONTINENT_NAME, REGION_NAME, CITY_NAME, TIMEZONE, LATITUDE, LONGITUDE, LOCATION]]
```

You'll notice that both yield the same error, even though the country database only supports the following fields: [`IP`, `COUNTRY_ISO_CODE`, `COUNTRY_NAME`, `CONTINENT_NAME`].

### PR DESCRIPTION
**Pull request by talevy**
Fixes #17683.

### CODE REVIEW

File: plugins/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpProcessorFactoryTests.java [hunk 1]
```diff
@@ -101,6 +101,7 @@ public void testBuildDbFile() throws Exception {
         assertThat(processor.getField(), equalTo("_field"));
         assertThat(processor.getTargetField(), equalTo("geoip"));
         assertThat(processor.getDbReader().getMetadata().getDatabaseType(), equalTo("GeoLite2-Country"));
+        assertThat(processor.getProperties(), sameInstance(GeoIpProcessor.Factory.DEFAULT_COUNTRY_PROPERTIES));
```
Comment (martijnvg) [inline plugins/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpProcessorFactoryTests.java]: maybe add a small test that fails that we fail if a city property is used for when the database type is country?

Comment (martijnvg) [inline plugins/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpProcessorFactoryTests.java]: ow I see this PR is WIP, so I'm guessing that you'll add it :)

Anyways this change does LGTM so far.

File: plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java [hunk 2]
```diff
@@ -233,24 +239,51 @@ public GeoIpProcessor doCreate(String processorTag, Map<String, Object> config)
...
+                        }
+                        properties.add(property);
+                    } catch (IllegalArgumentException e) {
+                        // TODO(talevy): add databaseType context to this exception message
                         throw newConfigurationException(TYPE, processorTag, "properties", "illegal field option [" + fieldName + "]. valid values are [" + Arrays.toString(Property.values()) + "]");
                     }
                 }
             } else {
-                properties = DEFAULT_PROPERTIES;
+                // TODO(talevy): figure out way to switch type without having to worry about valid type input (use if statements)
```
Comment (talevy) [inline plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java]: @martijnvg, you have no problems with this unnecessary (but safe) usage of the switch along with the exception throwing?

File: plugins/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpProcessorFactoryTests.java (hunk 1 above)
Comment (talevy) [inline plugins/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpProcessorFactoryTests.java]: yup 😄, but I'll add it now!

I also need to add another test for the error message that should be provided for each database type

Comment (talevy) [inline plugins/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpProcessorFactoryTests.java]: added below

File: plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java (hunk 2 above)
Comment (martijnvg) [inline plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java]: the code that exists now, doesn't bother me :)

**Comment by martijnvg** [comment]
LGTM

**Pull request merged**